src/unit_tests/wazuh_db/test_wdb_integrity.c

```diff
@@ -1526,6 +1527,156 @@ void test_wdbi_last_completion_step_fail(void **state)
 
 }
 
+// Test wdb_global_group_hash_cache
+
+void test_wdb_global_group_hash_cache_clear_success(void **state)
+{
+    wdb_global_group_hash_operations_t operation = WDB_GLOBAL_GROUP_HASH_CLEAR;
+    os_sha1 hexdigest;
+    int ret;
+
+    ret = wdb_global_group_hash_cache(operation, hexdigest);
+
+    assert_false(global_group_hash[0]);
+    assert_int_equal(ret, OS_SUCCESS);
+}
+
+void test_wdb_global_group_hash_cache_read_fail(void **state)
+{
+    wdb_global_group_hash_operations_t operation = WDB_GLOBAL_GROUP_HASH_READ;
+    os_sha1 hexdigest;
+    int ret;
+
+    ret = wdb_global_group_hash_cache(operation, hexdigest);
+
+    assert_false(global_group_hash[0]);
+    assert_int_equal(ret, OS_INVALID);
+}
+
+void test_wdb_global_group_hash_cache_write_success(void **state)
+{
+    wdb_global_group_hash_operations_t operation = WDB_GLOBAL_GROUP_HASH_WRITE;
+    os_sha1 hexdigest = "bd612e9ae2faf8a44b387eeb9f3d5a5e577c8c64";
+    int ret;
+
+    ret = wdb_global_group_hash_cache(operation, hexdigest);
+
+    assert_string_equal(global_group_hash, hexdigest);
+    assert_int_equal(ret, OS_SUCCESS);
+}
+
+void test_wdb_global_group_hash_cache_read_success(void **state)
+{
+    wdb_global_group_hash_operations_t operation = WDB_GLOBAL_GROUP_HASH_READ;
+    os_sha1 hexdigest;
+    int ret;
+
+    ret = wdb_global_group_hash_cache(operation, hexdigest);
+
+    assert_string_equal(hexdigest, "bd612e9ae2faf8a44b387eeb9f3d5a5e577c8c64");
+    assert_int_equal(ret, OS_SUCCESS);
+}
+
+void test_wdb_global_group_hash_cache_invalid_mode(void **state)
+{
+    wdb_global_group_hash_operations_t operation = 3;
+    os_sha1 hexdigest;
+    int ret;
+
+    expect_string(__wrap__mdebug2, formatted_msg, "Invalid mode for global group hash operation.");
+
+    ret = wdb_global_group_hash_cache(operation, hexdigest);
+
+    assert_int_equal(ret, OS_INVALID);
+}
+
+// Test wdb_get_global_group_hash
+
+void wdb_get_global_group_hash_read_success(void **state)
```

Comment:
This test isn't independent, it relies on previous execution to work properly. We can set the variables during the test, or modify the setup/teardown.